hack/update-codegen.sh

```diff
@@ -26,6 +26,12 @@ readonly GOMODCACHE="$(go env GOMODCACHE)"
 readonly GO111MODULE="on"
 readonly GOFLAGS="-mod=readonly"
 readonly GOPATH="$(mktemp -d)"
+readonly MIN_REQUIRED_GO_VER="1.19"
+
+if go version | perl -ne "exit 0 unless m{go version go([0-9]+.[0-9]+)}; exit 1 if (\$1 >= ${MIN_REQUIRED_GO_VER})"; then
```

Comment:
@shaneutt had to change it a little bit, the perl script should exit 0 in order to trigger the if statement, played a bit with the boolean logic to adjust for clarity after that. Also adjusted the regexp to work with first minor releases, which don't have the patch component:
```
❯ gvm use go1.19
Now using version go1.19
❯ go version
go version go1.19 linux/amd64
❯ gvm use go1.19.2
Now using version go1.19.2
❯ go version
go version go1.19.2 linux/amd64
```

Wdyt?

Reply:
My only thinking is that it might be a _little_ tricky for readers to understand at a glance because we're mixing an explicit success with an explicit failure _and_ "success" (`exit 0`) ultimately means "failure" in this if conditional. It's all kinda minor because at the end of the day, any time you're working with bash scripts running perl/sed/awk it always gets a little messy, but I do have a suggestion for your considerations which I think may help improve the "at a glance" readability here for future code readers:

```bash
function go_version_matches() {
    go version | perl -ne "exit 1 unless m{go version go([0-9]+.[0-9]+)}; exit 1 if (\$1 < 1.19)"
    return $?
}

if ! go_version_matches; then
    echo "Go v1.19 or later is required to run code generation"
    exit 1
fi
```

This way `exit 1` is just failure, and using a function it makes the conditional read a bit more clear. LMKWYT?

Reply:
Right, makes sense!